### DISCUSSION TITLE
Solis inverter Wifi dongle piggy backing the RS485

### DISCUSSION THREAD

**Discussion by mart-mac** [wills106/homeassistant-solax-modbus#192] [Q&A]
I’m waiting for some parts (RS485 adaptor) to arrive to give this integration on my Solis hybrid a try.  One thing I’d like to keep working is the Solis app.  

Do you think the RS485 would work piggy backing on the wifi dongle, i.e. soldering at couple of extra wires onto the dongle and running these to my RS485 adaptor.  Or am I going to blow it up?

**Comment by wills106** [comment]
I don't know how the WiFi sticks work.
But I don't know if the Waveshare sending commands to the Inverter at the same time as the WiFi Stick sending commands will confuse / overwhelm the Inverter or not?

They will be two active devices connected to your Inverter. I would expect the Inverter only expecting one master device and not two?

**Reply by rumpshaker204** [reply]
You won't blow it up but you will get spurious readings on soliscloud. I've been setting this up for the last month or so. A modbus can have only one master and by keeping the wifi dongle (I have the latest model 3) and then also connecting to HA you will end up with 2 masters and only one slave (the inverter). The effect of which is communication collisions which affect the soliscloud readings.

You could remove the wifi dongle completely and the problem is solved, however I wanted to keep the soliscloud integration (warranty and troubleshooting whilst I get all this HA done). The solution I found by reading a lot of threads was to spend £130 on a tS-735 made by idp.com - that allows multiple masters and one slave by buffering and queueing read requests from the masters.

I bought one of the solis connectors from ebay and soldered a few bits of wire. It's cost me about £200 and some long hours - all to get the inverter to not charge from the grid on demand. Steep learning curve ....

**Reply by alienatedsec** [reply]
@wills106 I know it was four months ago, but you got a gist of my setup for reference in #340 and I started documenting my efforts and configuration over here https://github.com/alienatedsec/solis-ha-modbus-cloud

@rumpshaker204 Waveshare is a gateway that can be queried by multiple devices
```
Support TCP server, TCP client, UDP mode, and UDP multicast. When acting as a TCP client, it also supports TCP server-side functions. Supports 30 TCP connections as a TCP server and 7 destination IPs as a TCP client.
```

And here is more about storage-type Modbus

```
Support Modbus gateway function, support Modbus RTU to Modbus TCP. It can support storage-type Modbus, and can automatically collect device data and store it; it also supports a non-storage mode Modbus gateway.
```

https://www.waveshare.com/wiki/RS485_TO_ETH_(B)#Documentation

**Reply by mart-mac** [reply]
Thanks for this info on your setup, looks just what I want, so I've brought myself a couple of these waveshare to eth and I'm going to give it a go.  Could you post some screen shots your you setting in vircom for the 2 waveshare's.  I'm know your stated some setting, but I'm totally new to all this and could do with as much help as possible. 

Thanks

**Reply by alienatedsec** [reply]
I wish I could share more. I even started it, and there were too many areas to anonymise. I can assure you that if you follow my [configuration.](https://github.com/alienatedsec/solis-ha-modbus-cloud#configuration), it will work as designed, and If you raise the issue in my repo, I can help with any further config.
